include/boost/geometry/srs/projections/impl/pj_generic_inverse.hpp

```diff
@@ -0,0 +1,145 @@
+// Boost.Geometry
+
+// Copyright (c) 2022, Oracle and/or its affiliates.
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+// This file is converted from PROJ, https://github.com/OSGeo/PROJ
+
+// Last updated version of proj: 9.0.0
+
+// Original copyright notice:
+
+/******************************************************************************
+ *
+ * Project:  PROJ
+ * Purpose:  Generic method to compute inverse projection from forward method
+ * Author:   Even Rouault <even dot rouault at spatialys dot com>
+ *
+ ******************************************************************************
+ * Copyright (c) 2018, Even Rouault <even dot rouault at spatialys dot com>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ ****************************************************************************/
+
+#include <algorithm>
+#include <cmath>
+
+#include <boost/geometry/util/math.hpp>
+
+/** Compute (lam, phi) corresponding to input (xy_x, xy_y) for projection P.
+ *
+ * Uses Newton-Raphson method, extended to 2D variables, that is using
+ * inversion of the Jacobian 2D matrix of partial derivatives. The derivatives
+ * are estimated numerically from the P->fwd method evaluated at close points.
+ *
+ * Note: thresholds used have been verified to work with adams_ws2 and wink2
+ *
+ * Starts with initial guess provided by user in lpInitial
+ */
+
+template <typename T, typename Parameters, typename Projection>
+void pj_generic_inverse_2d(T const& xy_x,
+                           T const& xy_y,
+                           Parameters const& par,
+                           Projection const& proj,
+                           T& lp_lat,
+                           T& lp_lon)
```

Comment:
Cool, new projection.

I cannot oversee all the details of course. Thanks for the work!